src/Factory.sol

```diff
@@ -21,13 +22,13 @@ contract Factory is ERC721, Ownable {
         address logic;
         address stakeContract;
         address interestModule;
+        bytes32 upgradeRoot;
     }
 
     mapping(address => bool) public isVault;
     mapping(uint256 => vaultVersionInfo) public vaultDetails;
 
-    uint256 public currentVaultVersion;
-    bool public factoryInitialised;
+    uint16 public currentVaultVersion;
```

Comment:
Nitpicking so you can ignore, but why not drop the 'current'?
Think vaultVersion is sufficient

Reply:
or latestVaultVersion